include/deal.II/base/vectorization.h

```diff
@@ -2707,6 +2707,25 @@ vectorized_transpose_and_store(const bool                    add_into,
 
 
 /**
+ * Relational operator == for VectorizedArray
+ *
+ * @relates VectorizedArray
+ */
+template <typename Number>
+inline DEAL_II_ALWAYS_INLINE
+bool
+operator == (const VectorizedArray<Number> &lhs,
```

Comment:
FYI. This function, along with [this](https://github.com/dealii/dealii/pull/4785/files#diff-3d00cdffd465e8ebfbffae5c7a742b52R602) and [this](https://github.com/dealii/dealii/pull/4785/files#diff-3d00cdffd465e8ebfbffae5c7a742b52R613) are only necessary to ensure that these checks
```
d == internal::NumberType<Number>::value(0.0)
```
are still valid for `Number == Tensor<1,dim,VectorizedArray<othernumber>>`. I don't know how this worked before but, after explicitly setting the RHS value in the assertion, this workaround was absolutely necessary.